internal/pr/creator.go

```diff
@@ -0,0 +1,246 @@
+	return err
+}
+
+func (c *Creator) createGitHubPR(ctx context.Context, branchName string, cfg config.RepositoryConfig) (string, error) {
+	repoName := extractRepoName(cfg.Name)
+
+	title := fmt.Sprintf("chore: add coverage tracking for %s", repoName)
+	body := c.generatePRBody(cfg)
+
+	newPR := &github.NewPullRequest{
+		Title:               github.String(title),
+		Head:                github.String(branchName),
+		Base:                github.String(c.baseBranch),
+		Body:                github.String(body),
+		MaintainerCanModify: github.Bool(true),
+	}
+
+	pr, _, err := c.client.PullRequests.Create(ctx, c.org, c.currentRepo, newPR)
+	if err != nil {
+		return "", err
+	}
+
+	// Add reviewers
+	if err := c.addReviewers(ctx, pr.GetNumber(), cfg.Owners); err != nil {
+		fmt.Printf("    ⚠️  Warning: failed to add reviewers: %v\n", err)
+	}
+
+	return pr.GetHTMLURL(), nil
+}
+
+func (c *Creator) addReviewers(ctx context.Context, prNumber int, owners []string) error {
+	reviewers := extractReviewers(owners)
+	if len(reviewers) == 0 {
+		return nil
+	}
+
+	// Separate individual reviewers from teams
+	var users []string
+	var teams []string
+
+	for _, reviewer := range reviewers {
+		if strings.Contains(reviewer, "/") {
+			parts := strings.Split(reviewer, "/")
+			if len(parts) == 2 {
+				teams = append(teams, parts[1])
+			}
+		} else {
+			users = append(users, reviewer)
+		}
+	}
+
+	reviewersRequest := github.ReviewersRequest{
+		Reviewers:     users,
+		TeamReviewers: teams,
+	}
+
+	_, _, err := c.client.PullRequests.RequestReviewers(ctx, c.org, c.currentRepo, prNumber, reviewersRequest)
+	return err
+}
+
+func (c *Creator) generatePRBody(cfg config.RepositoryConfig) string {
+	return fmt.Sprintf(prBodyTemplate, "`"+cfg.Name+"`")
+}
+
+// Helper functions
+
+func extractRepoName(fullName string) string {
+	parts := strings.Split(fullName, "/")
+	if len(parts) == 2 {
+		return filepath.Base(parts[1])
+	}
+	return filepath.Base(fullName)
+}
+
+func extractReviewers(owners []string) []string {
+	var reviewers []string
+	for _, owner := range owners {
+		reviewer := strings.TrimPrefix(owner, "@")
+		if reviewer != "" {
+			reviewers = append(reviewers, reviewer)
+		}
+	}
+	return reviewers
+}
+
+func formatList(items []string, emptyText string) string {
+	if len(items) == 0 {
+		return emptyText
+	}
+	var formatted []string
+	for _, item := range items {
+		formatted = append(formatted, "- `"+item+"`")
+	}
+	return strings.Join(formatted, "\n")
+}
+
+// Git helper functions
+
+// RunGitCommand executes a git command and returns the output and error
+func RunGitCommand(ctx context.Context, workDir string, args ...string) (string, error) {
+	cmd := exec.CommandContext(ctx, "git", args...)
+	cmd.Dir = workDir
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("git %s failed: %w\n%s", strings.Join(args, " "), err, string(output))
+	}
+	return string(output), nil
+}
```

Comment:
Have you considered using a module like `go-git`? That would help remove a dependency on the `git` CLI and allows you to work with native go structs/func instead of needing to parse output.